### PR TITLE
Fix PPO open env

### PR DESCRIPTION
**Pull request by vmoens**
closes the env as per #97 
Long-term fix should be to close all collectors and envs upon deletion, but we should be careful with that.

### CODE REVIEW

File: examples/ppo/ppo.py [hunk 1]
```diff
@@ -119,6 +119,8 @@ def make_args():
         recorder_rm = recorder
 
     recorder_rm.load_state_dict(create_env_fn.state_dict()["worker0"])
+    create_env_fn.close()
+
```
Comment (shagunsodhani) [inline examples/ppo/ppo.py:123]: Not sure if I follow but shouldnt we close the env after training?

Comment (vmoens) [reply examples/ppo/ppo.py:123]: not really
here what happens is that we might be normalizing observations on the fly using `VecNorm`
`VecNorm` uses shared memory to place the normalization stats (location and scale) such that they're available for every process.
If you have a recording environment, you must sync it with the others (i.e. give it access to the shared tensors). This is what `load_state_dict` does.
Now our `create_env_fn` is actually a `ParallelEnv` instance, and when we query its state dict, it must (1) run every process (which is not done upon creation of the object but only when needed, for instance when calling `reset` or `state_dict`) and (2) read the constant from one process.
Thing is that once we have queries `state_dict` we have also run the processes. Then they should be closed.


As a side note, I mentioned that `create_env_fn` is a parallel env. If you look at its `__init__`, it stores a (list of) picklable environment creation functions and stays idle until it is queried. When we create a data collector, we pass it a list of functions/objects to run on each subprocess. If it's a function, then each subprocess will create an environement from that function and use it to collect data. If it's an object (e.g. ParallelEnv) then each process will play with it to collect data. The ParallelEnv can be sent to the workers as it does not contain an instantiated env but only a pickable function that creates envs.
I made this very ugly sketch to show how creating envs in the collector works in the single and multi process case, hope you'll like it :p 

env dispatch 001

Comment (shagunsodhani) [reply examples/ppo/ppo.py:123]: Thanks for the detailed response! Appreciate it :)
